src/anomalib/callbacks/visualizer.py

```diff
@@ -0,0 +1,171 @@
+"""Visualizer Callback.
+
+This is assigned by Anomalib Engine internally.
+"""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from pathlib import Path
+from typing import Any, cast
+
+from lightning.pytorch import Callback, Trainer
+from lightning.pytorch.utilities.types import STEP_OUTPUT
+
+from anomalib.data.utils.image import save_image, show_image
+from anomalib.loggers import AnomalibWandbLogger
+from anomalib.loggers.base import ImageLoggerBase
+from anomalib.models import AnomalyModule
+from anomalib.utils.visualization import (
+    BaseVisualizer,
+    GeneratorResult,
+    VisualizationStep,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class _VisualizationCallback(Callback):
+    """Callback for visualization that is used internally by the Engine.
+
+    Args:
+        visualizers (BaseVisualizer | list[BaseVisualizer]):
+            Visualizer objects that are used for computing the visualizations. Defaults to None.
+        save (bool, optional): Save the image. Defaults to False.
+        root (Path | None, optional): The path to save the images. Defaults to None.
+        log (bool, optional): Log the images into the loggers. Defaults to False.
+        show (bool, optional): Show the images. Defaults to False.
+
+    Example:
+        >>> visualizers = [ImageVisualizer(), MetricsVisualizer()]
+        >>> visualization_callback = _VisualizationCallback(
+        ... visualizers=visualizers,
+        ...     save=True,
+        ...     root="results/images"
+        ... )
+
+        CLI
+        $ anomalib train --model Padim --data MVTec \
+            --visualization.visualizers ImageVisualizer \
+            --visualization.visualizers+=MetricsVisualizer
+        or
+        $ anomalib train --model Padim --data MVTec \
+            --visualization.visualizers '[ImageVisualizer, MetricsVisualizer]'
+
```

Comment:
is it possible to show an api example as well? Or is it not possible since it is handled by the engine?